bokeh/core/properties.py

```diff
@@ -1600,7 +1600,7 @@ def set_from_json(self, obj, json, models=None):
 
 class DataSpec(Either):
     def __init__(self, typ, default, help=None):
-        super(DataSpec, self).__init__(String, Dict(String, Either(String, typ)), typ, default=default, help=help)
+        super(DataSpec, self).__init__(String, Dict(String, Either(String, Instance(HasProps), typ)), typ, default=default, help=help)
```

Comment:
@justace yes probably, in this case you can pass a string name to `Instance` too, see e.g.

https://github.com/bokeh/bokeh/blob/master/bokeh/models/layouts.py#L112